### ISSUE TITLE
Does `stb_sprintf` not have format warnings? (gcc -Wformat)

### ISSUE THREAD

**Issue by aganm** [nothings/stb#1149]
If I do something such as

```
char buf[1024];
sprintf(buf, "%s", 1);
```
`gcc -Wformat` gives me this warning

```
warning: format ‘%s’ expects argument of type ‘char *’, but argument 3 has type ‘int’ [-Wformat=]
   19 | sprintf(buf, "%s", 1);                                                                           
      |               ~^   ~                                                                             
      |                |   |                                                                             
      |                |   int                                                                           
      |                char *   
```

If I use stb_sprintf
```
char buf[1024];
stbsp_sprintf(buf, "%s", 1);
```
No warnings.

Is there a way to get the same format warnings with stb_sprintf?

**Comment by turol** [comment]
You need to add attribute format to to printf prototype. Consult the GCC manual for details. https://gcc.gnu.org/onlinedocs/gcc-11.1.0/gcc/Common-Function-Attributes.html#index-format-function-attribute

**Comment by aganm** [comment]
Thanks!

**Comment by rygorous** [comment]
Fix for this is merged into the dev branch (off #1150) and will be in the next release.
